rosbridge_server/src/rosbridge_server/websocket_handler.py

```diff
@@ -96,9 +97,15 @@ def on_close(self):
         self.protocol.finish()
         rospy.loginfo("Client disconnected. %d clients total.", cls.clients_connected)
 
-    def send_message(self, message):
+    def send_message(self, message, topic=None):
+        if (self.stream._write_buffer_size==0):
+            self.blocked = []
         binary = type(message)==bson.BSON
-        IOLoop.instance().add_callback(partial(self.write_message, message, binary))
+        if topic == None or not binary:
```

Comment:
Forgive me if these comments are too pedantic.  Relatively new to community PRs and code reviews =)

```if topic is None or not binary``` would be more Pythonic.